app/src/main/java/at/aau/edu/wizards/gameModel/GameModelDataListener.kt

```diff
@@ -0,0 +1,91 @@
+package at.aau.edu.wizards.gameModel
+
+class GameModelDataListener(private val parent: GameModel, private val rules: GameModelRules) :
+    GameModelDataListenerInterface {
+    private var activePlayer = 0
+    private var numberOfPlayers = 0
+    private var currentHands = ArrayList<GameModelCard>()
+    private var activeTrump = GameModelCard(buildString {
+        append(0.toChar())
+        append(0.toChar())
+        append(0.toChar())
+    })
+    private var currentGuess = ArrayList<Guess>()
+    private var totalScores = ArrayList<Score>()
+
+    data class Guess(val guess: Int, val playerId: Int)
+    data class Score(val score: Int, val playerId: Int)
+
+    override fun getActivePlayer(): Int {
```

Comment:
properties with public read and private set can be written

```
var property: Int = 0
    private set
``

Reply:
Changed all private variables with public getters to public variables with private setters in new implementation.